bindings/go/oci/resolver/url/resolver.go

```diff
@@ -57,6 +57,14 @@ func (resolver *CachingResolver) Reference(reference string) (fmt.Stringer, erro
 	return registry.ParseReference(reference)
 }
 
+func (resolver *CachingResolver) Ping(ctx context.Context) error {
+	r, err := remote.NewRegistry(resolver.baseURL)
```

Comment:
isnt this registry something we can create when we create the resolver from the base url? Also im wondering if you tested this against a real registry, because from what I can see you never pass the credentials / the http Client anywhere.

I was expecting an integration test to verify this against ghcr and something like this:

```
remote.Registry{
		RepositoryOptions: remote.RepositoryOptions{
			PlainHTTP: resolver.plainHTTP,
			Client:    repo.Client,
		},
	}
```

instead of default construction

Reply:
I think it's actually just this

```go
	
	r.PlainHTTP = resolver.plainHTTP
	if resolver.baseClient != nil {
		r.Client = resolver.baseClient
	}
```

The client is there. This might be nice. :)